Include number of Identical Files in QA stats and meter

tw4l opened the pull request:
Fixes #1831 

This PR adds Identical Files to the QA Page Match Analysis meter bars. To do this, the backend calculates the number of non-HTML pages once and includes it under the key `Files` in each of the `screenshotMatch` and `textMatch` QA stats return arrays.

The backend additionally removes the file count from "No Data" to prevent these from being counted twice.

## Screenshots

### QA run complete, all pages analyzed

<img width="965" alt="Screenshot 2024-06-04 at 2 08 13 PM" src="https://github.com/webrecorder/browsertrix/assets/6758804/915ddf5a-22e9-446c-9455-b6c2af43284f">

### QA run stopped prematurely, pages with no data

<img width="970" alt="Screenshot 2024-06-04 at 2 07 42 PM" src="https://github.com/webrecorder/browsertrix/assets/6758804/7955130e-69a2-424b-84ab-78825a5193df">

<img width="973" alt="Screenshot 2024-06-04 at 2 07 53 PM" src="https://github.com/webrecorder/browsertrix/assets/6758804/d2f70b7a-d640-4cbe-aa1f-1744dee62d11">

<img width="965" alt="Screenshot 2024-06-04 at 2 07 59 PM" src="https://github.com/webrecorder/browsertrix/assets/6758804/72c8b7f4-4e8d-46ba-b77d-51fc4ab2c69a">

<img width="965" alt="Screenshot 2024-06-04 at 2 08 35 PM" src="https://github.com/webrecorder/browsertrix/assets/6758804/82fab3a4-e62c-4cf0-b416-fcb52a283b24">

tw4l committed:
Include file count in QA stats scores

Calculate number of identical files once but return as part of
each key in the QA stats output to help with graphing.

tw4l committed:
Exclude identical files from No data key

tw4l committed:
Frontend fixups

tw4l committed:
Test for Files in QA stats response

tw4l committed:
Fix Good Match label

tw4l commented on frontend/src/pages/org/archived-item-detail/ui/qa.ts:
```diff
@@ -653,12 +658,12 @@ export class ArchivedItemDetailQA extends TailwindElement {
                           : idx === qaStatsThresholds.length - 1
                             ? `>=${threshold ? +threshold.lowerBoundary * 100 : 0}%`
-                            : `${threshold ? +threshold.lowerBoundary * 100 : 0}-${+qaStatsThresholds[idx + 1].lowerBoundary * 100}%`}
+                            : `${threshold ? +threshold.lowerBoundary * 100 : 0}-100%`}
```
This isn't working as expected, will need tweaking

Shrinks99 commented:
Looks good (other than the inconsistencies above which have paths to fixing!)

Any chance the inconsistent bar rounding between this and the org dashboard meters could be fixed?? 👀 

The bar should have the same corner radius as the container, currently it does not.

<img width="571" alt="Screenshot 2024-06-04 at 4 41 12 PM" src="https://github.com/webrecorder/browsertrix/assets/5672810/128846cc-4d6c-4223-8c31-e86745859f9b">

<img width="502" alt="Screenshot 2024-06-04 at 4 42 30 PM" src="https://github.com/webrecorder/browsertrix/assets/5672810/aa233463-0c45-486f-baaa-f30855080117">

Shrinks99 commented:
Oh actually one thing that should be fixed... The use of the exclamation triangle is not inline with how we use the status icons.  These aren't used for stopped statuses anywhere else and are typically reserved for severe errors.

<img width="824" alt="Screenshot 2024-06-04 at 5 07 33 PM" src="https://github.com/webrecorder/browsertrix/assets/5672810/8a1c279a-3fbd-4f6c-9162-2bd343e188f9">

I think I would remove the tooltip too??  It's putting into text information that should be communicated by the graph's UI & status icon?  The `dash-square` icon everywhere else in the app also indicates an item that is incomplete but still has data.  Based on the graph, seems fairly intuitive to learn?

tw4l commented:
@Shrinks99 I actually wonder if we can do away with that icon (and tooltip) given that it's a bit redundant in context.

To show you what I mean, here's it with `dash-square`:

<img width="969" alt="Screenshot 2024-06-05 at 11 09 58 AM" src="https://github.com/webrecorder/browsertrix/assets/6758804/acf24e03-edaf-4725-95bd-246d4c3a2d4d">

And entirely removed:

<img width="971" alt="Screenshot 2024-06-05 at 11 11 22 AM" src="https://github.com/webrecorder/browsertrix/assets/6758804/c979e864-af94-479a-b587-8a5b7b3453de">

Given the two other dash-squares in close proximity for a stopped analysis run, I think I prefer it with no icon at all next to the page count. What do you think?

tw4l commented:
> Any chance the inconsistent bar rounding between this and the org dashboard meters could be fixed?? 👀

Yes, and we probably should, but this is the kind of finicky thing that might take a surprising amount of time to get right. The existing btrix-divided-meter-bar component we use in the dashboard doesn't really fit here in a plug-and-play way. I'd suggest we separate this into its own issue to avoid this PR from dragging out too long, since it's a bit of scope creep anyway.

Shrinks99 commented:
> Given the two other dash-squares in close proximity for a stopped analysis run, I think I prefer it with no icon at all next to the page count. What do you think?

+1, that's kinda where I was going with my thoughts there, sorry if unclear 🙃

tw4l committed:
Implement code review changes

- Remove triange exclamation icon next to pages analyzed count
- Fix calculation of upper boundary percentage in bars

tw4l commented:
Issue created for rounding the bars: https://github.com/webrecorder/browsertrix/issues/1851

tw4l commented:
@Shrinks99 Latest changes pushed to dev if you want to take another look

tw4l commented:
@ikreymer want to take a quick look at backend changes before merging?

emma-sg committed:
ensure text can be translated

emma-sg reviewed (approved):
LGTM!
